src/codegen/forward_demand.jl

```diff
@@ -105,6 +105,7 @@ function forward_diff_uncached!(ir::IRCode, interp::AbstractInterpreter, irsv::I
             Δbacking = insert_node!(ir, ssa, NewInstruction(Expr(:splatnew, widenconst(tup_typ), Δbacking), tup_typ_typ.val))
         end
         tangentT = Core.Compiler.apply_type_tfunc(Const(ChainRulesCore.Tangent), newT, tup_typ_typ).val
+        # TODO do we need to make sure this inserts right 
```

Comment:
What does this mean?

Reply:
I think we wanted to make sure that it was inserting the node in the right spot in the IR (specifically in the presence of control flow)

Reply:
Ok seems unreleated to this PR

```suggestion
```